components/core/src/clp_s/timestamp_parser/TimestampParser.cpp

```diff
@@ -985,8 +1207,80 @@ auto parse_timestamp(
             std::chrono::duration_cast<std::chrono::nanoseconds>(time_point.time_since_epoch())
                     .count()
     };
-    return {epoch_nanoseconds, pattern};
+    return {epoch_nanoseconds, returned_pattern};
 }
 
 // NOLINTEND(readability-function-cognitive-complexity)
+
+[[nodiscard]] auto search_known_timestamp_patterns(
+        std::string_view timestamp,
+        std::vector<TimestampPattern> const& patterns,
+        std::string& generated_pattern
+) -> ystdlib::error_handling::Result<std::pair<epochtime_t, std::string_view>> {
+    for (auto const& pattern : patterns) {
+        auto const result{parse_timestamp(timestamp, pattern, generated_pattern)};
+        if (false == result.has_error()) {
+            return result.value();
+        }
+    }
+    return ErrorCode{ErrorCodeEnum::IncompatibleTimestampPattern};
+}
+
+auto get_default_date_time_timestamp_patterns()
+        -> ystdlib::error_handling::Result<std::vector<TimestampPattern>> {
+    std::vector<TimestampPattern> timestamp_patterns;
+    timestamp_patterns.reserve(cDefaultDateTimePatterns.size());
+    for (auto const& pattern : cDefaultDateTimePatterns) {
+        auto timestamp_pattern{YSTDLIB_ERROR_HANDLING_TRYX(TimestampPattern::create(pattern))};
+        timestamp_patterns.emplace_back(std::move(timestamp_pattern));
+    }
+    return timestamp_patterns;
+}
+
+auto get_default_numeric_timestamp_patterns()
+        -> ystdlib::error_handling::Result<std::vector<TimestampPattern>> {
+    std::vector<TimestampPattern> timestamp_patterns;
+    timestamp_patterns.reserve(cDefaultNumericPatterns.size());
+    for (auto const& pattern : cDefaultNumericPatterns) {
+        auto timestamp_pattern{YSTDLIB_ERROR_HANDLING_TRYX(TimestampPattern::create(pattern))};
+        timestamp_patterns.emplace_back(std::move(timestamp_pattern));
+    }
+    return timestamp_patterns;
+}
+
+auto get_all_default_timestamp_patterns()
+        -> ystdlib::error_handling::Result<std::vector<TimestampPattern>> {
+    std::vector<TimestampPattern> timestamp_patterns;
+    timestamp_patterns.reserve(cDefaultNumericPatterns.size() + cDefaultDateTimePatterns.size());
+    for (auto const& pattern : cDefaultNumericPatterns) {
+        auto timestamp_pattern{YSTDLIB_ERROR_HANDLING_TRYX(TimestampPattern::create(pattern))};
+        timestamp_patterns.emplace_back(std::move(timestamp_pattern));
+    }
+    for (auto const& pattern : cDefaultDateTimePatterns) {
+        auto timestamp_pattern{YSTDLIB_ERROR_HANDLING_TRYX(TimestampPattern::create(pattern))};
+        timestamp_patterns.emplace_back(std::move(timestamp_pattern));
+    }
```

Comment:
```suggestion
    auto const date_time_patterns{
            YSTDLIB_ERROR_HANDLING_TRYX(get_default_date_time_timestamp_patterns())
    };
    auto const numeric_patterns{
            YSTDLIB_ERROR_HANDLING_TRYX(get_default_numeric_timestamp_patterns())
    };
    timestamp_patterns.reserve(date_time_patterns.size() + numeric_patterns.size());
    timestamp_patterns.insert(
            timestamp_patterns.cend(),
            date_time_patterns.cbegin(),
            date_time_patterns.cend()
    );
    timestamp_patterns
            .insert(timestamp_patterns.cend(), numeric_patterns.cbegin(), numeric_patterns.cend());
```
I assume it's not used in a critical path, so how about doing it this way to avoid duplication?

Reply:
Sure, likely it is fine.